newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalFileResourceMover.js

```diff
@@ -65,21 +69,43 @@ const downloadBlobToLocalFile = async (
   );
 };
 
+// This mover can be used for both public URLs and Cloud project resources.
 export const moveUrlResourcesToLocalFiles = async ({
   project,
   fileMetadata,
   onProgress,
+  authenticatedUser,
 }: Options) => {
   if (!fs || !ipcRenderer) throw new Error('Unsupported');
 
   // Get all resources to download.
   const resourcesManager = project.getResourcesManager();
   const allResourceNames = resourcesManager.getAllResourceNames().toJSArray();
-  const resourceToFetchNames = allResourceNames.filter(resourceName => {
+  let tokenForPrivateGameTemplateAuthorization = null;
```

Comment:
Same as before, see if you can reuse a same function that "fetch the token if necessary".

It's a bit painful for the reader to read these 23 lines one by one just at the end to understand "oh this is just getting the token if needed, ok cool but this interrupted by understanding of the rest of the mover for such a small detail" ;)

Reply:
agreed, it's the same logic as well. I'll refactor that